pkg/asset/agent/image/agentimage.go

```diff
@@ -103,29 +108,75 @@ func (a *AgentImage) updateIgnitionImg(ignition []byte) error {
 		return err
 	}
 
-	ignitionImgPath := filepath.Join(a.tmpPath, "images", "ignition.img")
-	fi, err := os.Stat(ignitionImgPath)
+	// Defining a struct for the ignition info
+	type IgnInfo struct {
+		File   string `json:"file"`
+		Length int64  `json:"length"`
+		Offset int64  `json:"offset"`
+	}
+	var ignInfo IgnInfo
+
+	// Reading the ignition details from igninfo.json file into the struct
+	ignInfoJSONPath := filepath.Join(a.tmpPath, "coreos", "igninfo.json")
+	ignInfoJSONData, err := os.ReadFile(ignInfoJSONPath)
 	if err != nil {
+		logrus.Warnf("Failed to read json %s", ignInfoJSONPath)
+		return err
+	}
+	if err := json.Unmarshal(ignInfoJSONData, &ignInfo); err != nil {
+		logrus.Warnf("Failed to umarshal json: %s", err)
 		return err
 	}
 
-	// Verify that the current compressed ignition archive does not exceed the
-	// embed area (usually 256 Kb)
-	if len(ignitionBuff) > int(fi.Size()) {
-		return fmt.Errorf("ignition content length (%d) exceeds embed area size (%d)", len(ignitionBuff), fi.Size())
+	// Fetching the ignition details from the struct
+	ignImagePath := filepath.Join(a.tmpPath, ignInfo.File)
+	ignStartOffset := ignInfo.Offset
```

Comment:
@andfasano ,yeah for s390x the offset length differs. Below is THE reference on how igninfo.json looks for s390x 

For reference the ignition info file looks like this on s390x:

`{
  "file": "images/cdboot.img",
  "length": 262144,
  "offset": 66086252
}`

Reply:
I feel like we're replicating here something that is already implemented in https://github.com/openshift/assisted-image-service/blob/main/pkg/overlay/overlay.go

Even better though would be if we had some higher-level API for editing the Ignition. We used to use https://github.com/openshift/assisted-image-service/blob/main/pkg/isoeditor/stream.go#L29 but we switched to exploding the ISO and repacking it ourselves because we wanted to add another image. Nevertheless, we are using code from assisted-image-service to do that and I would prefer to see us add APIs (if necessary) there to do it, so that the hosted assisted service can also benefit.

Reply:
Modified as discussed